### PR TITLE
Treat cancel button for new model pages

### PR DESCRIPTION
**Pull request by gusaiani**
#### For new model pages

1. `+ New`
2. `Model`
3. Either option
4. `Cancel`

If `Yes` in confirmation modal, you should be taken to the home page.
If `No` in confirmation modal, you should be back creating the new model.

#### For saved models

1. Go to an existing model
2. `Explore`
3. `Edit query definition`
4. `Cancel`

Nothing should be different from how the app works today.
You should be taken back to the model page, no confirmation modal.

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/DatasetEditor/DatasetEditor.jsx [hunk 1]
```diff
@@ -415,11 +422,17 @@ function DatasetEditor(props) {
           />
         }
         buttons={[
-          <Button
-            key="cancel"
-            onClick={handleCancel}
-            small
-          >{t`Cancel`}</Button>,
+          dataset.isSaved() ? (
+            <Button small onClick={handleCancelEdit}>{t`Cancel`}</Button>
```
Comment (daltojohnso) [inline frontend/src/metabase/query_builder/components/DatasetEditor/DatasetEditor.jsx:426]: does this need a `key`?

Comment (gusaiani) [reply frontend/src/metabase/query_builder/components/DatasetEditor/DatasetEditor.jsx]: Yes, thanks. Updated.

File: frontend/src/metabase/query_builder/actions/ui.js [hunk 2]
```diff
@@ -13,6 +14,10 @@ export const setUIControls = createAction(SET_UI_CONTROLS);
 export const RESET_UI_CONTROLS = "metabase/qb/RESET_UI_CONTROLS";
 export const resetUIControls = createAction(RESET_UI_CONTROLS);
 
+export const onCancelCreateNewModel = () => async dispatch => {
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/actions/ui.js:17]: nit: maybe add it to `actions/models`?

**Review by ranquild — COMMENTED** [review]
I find this modal a bit confusing. Maybe change `Cancel` to `No`?

Screenshot 2022-10-11 at 14 29 37

**Review by ranquild — APPROVED** [review]
Approving the PR since I have concerns about the UX only

**Comment by gusaiani** [comment]
@ranquild Thanks for the review.

I agree with you about the Confirm modal "Cancel" vs "No" button label.

As the Confirm modal touches several other use cases, I will open a separate PR for it.

**Pull request merged**